packages/protocol/contracts/common/UsingPrecompiles.sol

```diff
@@ -25,38 +29,19 @@ contract UsingPrecompiles {
     require(aDenominator != 0 && bDenominator != 0);
     uint256 returnNumerator;
     uint256 returnDenominator;
-    // solhint-disable-next-line no-inline-assembly
-    assembly {
-      let newCallDataPosition := mload(0x40)
-      mstore(0x40, add(newCallDataPosition, calldatasize))
-      mstore(newCallDataPosition, aNumerator)
-      mstore(add(newCallDataPosition, 32), aDenominator)
-      mstore(add(newCallDataPosition, 64), bNumerator)
-      mstore(add(newCallDataPosition, 96), bDenominator)
-      mstore(add(newCallDataPosition, 128), exponent)
-      mstore(add(newCallDataPosition, 160), _decimals)
-      let success := staticcall(
-        1050, // estimated gas cost for this function
-        0xfc,
-        newCallDataPosition,
-        0xc4, // input size, 6 * 32 = 192 bytes
-        0,
-        0
-      )
-
-      let returnDataSize := returndatasize
-      let returnDataPosition := mload(0x40)
-      mstore(0x40, add(returnDataPosition, returnDataSize))
-      returndatacopy(returnDataPosition, 0, returnDataSize)
 
-      switch success
-        case 0 {
-          revert(returnDataPosition, returnDataSize)
-        }
-        default {
-          returnNumerator := mload(returnDataPosition)
-          returnDenominator := mload(add(returnDataPosition, 32))
-        }
+    bool success;
+    bytes memory result;
+    (success, result) = FRACTION_MUL_EXP.staticcall(
+      abi.encodePacked(aNumerator, aDenominator, bNumerator, bDenominator, exponent, _decimals)
+    );
+    require(
+      success,
+      "UsingPrecompiles :: fractionMulExp Unsuccessful invocation of fraction exponent"
+    );
+    assembly {
+      returnNumerator := mload(add(result, 32))
```

Comment:
Actually, are you sure this is right? Doesn't mload take the start, which in this case would be `result` and not `result + 32`?

Reply:
@asaj AFAIK, for length >= 32 bytes (which is the case with uint256), bytes use the first 32 bytes as length. This means that we need to "forward" 32 bytes from the beginning, hence the `add(result, 32))`. 
Regardless I double checked it and double tested this via the existing unit tests, and this is the correct setup :)

https://solidity.readthedocs.io/en/v0.5.13/miscellaneous.html#bytes-and-string

Reply:
@Perseverance is right... 
I've tested in remix, here's the code i've used https://gist.github.com/mcortesi/b35caa87abc0376ce58e7a4df11458a6

My doubt was not about that bytes first 32 bytes are the length, but the difference between making an assembly call or a call through the address... it seems they return things differently.

When assembly, there's no extra length field.. the total length is 0x40 bytes. But when we do the call with address.staticcall the returned bytes has a size 0x60 bytes.